pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go

```diff
@@ -253,3 +266,60 @@ func TestPartitionedTenantStreamingEndToEnd(t *testing.T) {
 		testTenantStreaming(t, false /* withInitialScan */)
 	})
 }
+
+func TestTenantStreamingProducerJobTimedOut(t *testing.T) {
```

Comment:
This seems like a fine test, but it isn't clear to me it tests for the bug that I observed. That is, if you run this test with the bug still in place, does this test fail?

If it does, great!  If not, this still looks like a useful test, but perhaps we additionally want something like I wrote here:

https://github.com/stevendanna/cockroach/commit/c51ca38408ed1cc285f7d4fd50060763510ab82b